app/bundles/EmailBundle/Entity/StatRepository.php

```diff
@@ -540,24 +541,43 @@ public function findContactEmailStats($leadId, $emailId)
     }
 
     /**
-     * @param $contacts
-     * @param $emailId
+     * @param      $contacts
+     * @param      $emailId
+     * @param bool $organizeByContact
      *
-     * @return mixed
+     * @return array|mixed|string
      */
-    public function checkContactsSentEmail($contacts, $emailId)
+    public function checkContactsSentEmail($contacts, $emailId, $organizeByContact = false)
     {
+        if (is_array($contacts)) {
+            $contacts = implode(',', $contacts);
+        }
+
         $query = $this->getEntityManager()->getConnection()->createQueryBuilder();
         $query->from(MAUTIC_TABLE_PREFIX.'email_stats', 's');
         $query->select('id, lead_id')
-        ->where('s.email_id = :email')
-        ->andWhere('s.lead_id in (:contacts)')
+            ->where('s.email_id = :email')
+            ->andWhere('s.lead_id in (:contacts)')
             ->andWhere('is_failed = 0')
-        ->setParameter(':email', $emailId)
-        ->setParameter(':contacts', $contacts);
+            ->setParameter(':email', $emailId)
+            ->setParameter(':contacts', $contacts, Connection::PARAM_INT_ARRAY)
+            ->groupBy('lead_id');
 
         $results = $query->execute()->fetch();
 
+        if ($organizeByContact) {
```

Comment:
Instead of adding another flag to this method, can you create a new method that calls this method and then modify the results as you need?